tests-legacy/Unit/Classes/ToolsCoreTest.php

```diff
@@ -369,6 +369,48 @@ public function testStrReplaceFirst($search, $replace, $subject, $cur, $expected
         $this->assertEquals($expected, Tools::StrReplaceFirst($search, $replace, $subject, $cur));
     }
 
+    /**
+     * @param string $passwordGenerated
+     * @param string $expectedPassword
+     *
+     * @dataProvider passwodGenProvider
+     */
+    public function testPasswdGen($expectedPassword, $passwordGenerated)
+    {
+        $this->assertRegExp($expectedPassword, $passwordGenerated, 'The password generated ' . $expectedPassword . ' no match with ' . $expectedPassword);
+    }
+
+    public function passwodGenProvider()
+    {
+        return [
+            '/^(\d{8})$/' => Tools::passwdGen(),
```

Comment:
It's a data provider, so the return should be 
```suggestion
            [
                '/^(\d{8})$/',
                Tools::passwdGen(),
            ],
```
and not

```suggestion
            '/^(\d{8})$/' => Tools::passwdGen(),
```